### PR TITLE
core: async + optimized TableSchema::from_rows

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Move all loops in upsert_rows to a blocking_spawn
- Optimize TableSchema::from_rows from LC^2 complexity to LC

To achieve that we move rows into the main api handler function and then wrap it in an Arc<> for safe cross-thread read-only operations.

We also do that when querying a table and creating the table schema. We unwrap the Arc at the end (which can't fail as we won't have more than 1 weak reference at this time)

## Risk

Tested locally

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/src/databases/table.rs [hunk 1]
```diff
@@ -228,41 +232,63 @@ impl LocalTable {
...
-                    "Row {} has a key '{}' that contains uppercase characters",
-                    row_index,
-                    key
-                ))?,
-                None => (),
-            }
+        let rows = Arc::new(rows);
+
+        let mut now = utils::now();
+        // Validate that all rows keys are lowercase. We run it in a spawn_blocking since it is CPU
```
Comment (fontanierh) [inline core/src/databases/table.rs:241]: Why not rayon for this one ?

File: front/lib/api/tables.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ import logger from "@app/logger/logger";
 
 import type { DataSourceResource } from "../resources/data_source_resource";
 
-const MAX_TABLE_COLUMNS = 512;
+const MAX_TABLE_COLUMNS = 52000;
```
Comment (fontanierh) [inline front/lib/api/tables.ts:29]: Is this on purpose ?

**Review by fontanierh — APPROVED** [review]
nice

File: core/src/databases/table.rs (hunk 1 above)
Comment (spolu) [reply core/src/databases/table.rs:241]: This one is very fast already with wide tables

File: front/lib/api/tables.ts (hunk 2 above)
Comment (spolu) [reply front/lib/api/tables.ts:29]: Nope cleaning

File: core/src/databases/table.rs (hunk 1 above)
Comment (spolu) [reply core/src/databases/table.rs:241]: or even deep tables... it's fast so no need for rayon but good hygiene to not block the pool

**Comment by philipperolet** [comment]
LVGTM :muscle:

**Pull request merged**